odoo/addons/test_access_rights/tests/test_feedback.py

```diff
@@ -398,6 +398,40 @@ def test_warn_company_access(self):
         ):
             p.with_user(self.user).val
 
+    def test_warn_company_access_multi_record(self):
+        """ because of prefetching, read() goes through a different codepath
+        to apply rules
+        """
+
+        company1_id = self.env['res.company'].create({'name': 'Brosse Inc.'})
+        company2_id = self.env['res.company'].create({'name': 'Brosse Inc. 2'})
+        record_1 = self.env["test_access_right.some_obj"].create({"val": 5, "company_id": company1_id.id})
+        record_2 = self.env["test_access_right.some_obj"].create({"val": 6, "company_id": company2_id.id})
+        self.user.sudo().company_ids = [Command.link(company1_id.id)]
+        self.user.sudo().company_ids = [Command.link(company2_id.id)]
+        self._make_rule('rule 0', "[('company_id', 'in', [False])]", attr='read')
+        with self.debug_mode(), self.assertRaises(AccessError) as ctx:
+            records = record_1 + record_2
+            self.env.invalidate_all()
+            _ = records.with_user(self.user).read(["val"])
+        self.maxDiff = None
+        self.assertEqual(
+            ctx.exception.args[0],
+            """Uh-oh! Looks like you have stumbled upon some top-secret records.
+
+Sorry, %s (id=%s) doesn't have 'read' access to:
+- %s, %s (%s: %s, company=%s)
+ - %s, %s (%s: %s, company=%s)
+
+Blame the following rules:
+- rule 0
+
+If you really, really need access, perhaps you can win over your friendly administrator with a batch of freshly baked cookies.
+
+Note: this might be a multi-company issue. Switching company may help - in Odoo, not in real life!"""
+        % (self.user.name, self.user.id, record_1._description, record_1.display_name, record_1._name, record_1.id, record_1.sudo().company_id.display_name, record_2._description, record_2.display_name, record_2._name, record_2.id, record_2.sudo().company_id.display_name))
```

Comment:
Suggestion to clean a little bit the test (variable renaming, update comments, remove noise, f-string, ...):
```suggestion
        """ Test that AccessError handle correctly several companies """
        company_1, company_2 = self.env['res.company'].create([
            {'name': 'Brosse Inc.'},
            {'name': 'Brosse Inc. 2'},
        ])
        records = self.env["test_access_right.some_obj"].create([
            {"val": 1, "company_id": company_1.id},
            {"val": 2, "company_id": company_2.id},
        ])
        record_1, record_2 = records
        self.user.sudo().company_ids = [Command.link(company_1.id), Command.link(company_2.id)]
        self._make_rule('rule 0', "[('company_id', '=', False)]", attr='read')
        self.env.invalidate_all()
        with self.debug_mode(), self.assertRaises(AccessError) as ctx:
            _ = records.with_user(self.user).read(["val"])
        self.assertEqual(
            ctx.exception.args[0],
            f"""Uh-oh! Looks like you have stumbled upon some top-secret records.

Sorry, {self.user.name} (id={self.user.id}) doesn't have 'read' access to:
- {record_1._description}, {record_1.display_name} ({record_1._name}: {record_1.id}, company={record_1.company_id.display_name})
- {record_2._description}, {record_2.display_name} ({record_2._name}: {record_2.id}, company={record_2.company_id.display_name})

Blame the following rules:
- rule 0

If you really, really need access, perhaps you can win over your friendly administrator with a batch of freshly baked cookies.

Note: this might be a multi-company issue. Switching company may help - in Odoo, not in real life!""")
```


Also you should move `self.maxDiff = None` in the `setUpClass` (`cls.maxDiff = None`)